### PR TITLE
Filter mentor inbox

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This builds on #64, which itself builds on #63.

### CODE REVIEW

File: app/javascript/components/mentoring/mentor_inbox.jsx [hunk 1]
```diff
@@ -11,6 +13,16 @@ export function reducer(state, action) {
...
       }
+    case 'sort.changed':
+      return {
+        ...state,
+        query: { ...state.query, sort: action.payload.sort },
+      }
+    case 'filter.changed':
+      return {
+        ...state,
+        query: { ...state.query, filter: action.payload.filter },
```
Comment (kntsoriano) [inline app/javascript/components/mentoring/mentor_inbox.jsx]: Should filtering reset the page?

**Review by kntsoriano — APPROVED** [review]
Looks great!

File: app/javascript/components/mentoring/mentor_inbox.jsx (hunk 1 above)
Comment (ErikSchierboom) [reply app/javascript/components/mentoring/mentor_inbox.jsx]: Absolutely! Good catch

File: app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb [hunk 2]
```diff
@@ -48,6 +48,12 @@ def index
     ]
 
     results = results.select { |c| c[:trackId] == params[:track].to_i } if params[:track].present?
+    if params[:filter].present?
+      results = results.select do |c|
+        c[:exerciseTitle].downcase.include?(params[:filter].downcase) ||
```
Comment (SleeplessByte) [inline app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb:53]: FYI, you might want to have done something like:

```ruby
parts = params[:filter].downcase.split(' ').map(&:strip)

parts.all? do |part|
  c[:exerciseTitle].downcase.include?(part) || c[:menteeHandle].downcase.include?(part)
end
```

Which is more in line with how people "think" search works.

File: app/javascript/components/mentoring/mentor_inbox.jsx [hunk 3]
```diff
@@ -17,6 +18,11 @@ function reducer(state, action) {
...
         query: { ...state.query, sort: action.payload.sort },
       }
+    case 'filter.changed':
+      return {
+        ...state,
+        query: { ...state.query, filter: action.payload.filter, page: 1 },
+      }
     default:
       if (process.env.NODE_ENV === 'development') {
         throw 'Unknown action type'
```
Comment (SleeplessByte) [inline app/javascript/components/mentoring/mentor_inbox.jsx:28]: Always throw errors, and add debuggable information to the error (preferably actionable).

```js
new Error(`Unknown action type: ${action.type}`)
```

Comment (ErikSchierboom) [reply app/javascript/components/mentoring/mentor_inbox.jsx:28]: I've fixed this in #78

File: app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb (hunk 2 above)
Comment (ErikSchierboom) [reply app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb:53]: I've fixed this in #78

**Pull request merged**